bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D2_50/D2_50.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.enocean.internal.eep.D2_50;
+
+import static org.openhab.binding.enocean.internal.EnOceanBindingConstants.*;
+
+import java.util.function.Function;
+
+import org.openhab.binding.enocean.internal.Helper;
+import org.openhab.binding.enocean.internal.eep.Base._VLDMessage;
+import org.openhab.binding.enocean.internal.messages.ERP1Message;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ *
+ * @author Daniel Weber - Initial contribution
+ */
+public class D2_50 extends _VLDMessage {
+
+    protected final byte mtMask = (byte) 0xf0;
+    protected final byte MT_REMOTE_TRANSMISSION_REQUEST = 0x00;
+    protected final byte MT_CONTROL = 0x20;
+    protected final byte MT_BASIC_STATUS = 0x40;
+    protected final byte MT_EXTENDED_STATUS = 0x60; // not yet implemented
+
+    protected final byte rmtMask = (byte) 0x0f;
+    protected final byte RMT_BASIC_STATUS = 0x00;
+    protected final byte RMT_EXTENDED_STATUS = 0x01; // not yet implemented
+
+    protected final byte DOMC_NOACTION = 0x0f;
+    protected final byte CONTROL_NOACTION = 0;
+    protected final byte TMOC_NOACTION = 127;
+    protected final byte TMOC_ACTIVATE = (byte) 0xff;
+    protected final byte THRESHOLD_NOACTION = 127;
```

Comment:
If the fields are immutable and not defined in the constructor, please them static as well.

Reply:
Sorry was my fault, thought Javas final equals c# const. Changed these field to static final.